### PR TITLE
qa: Add getdescriptorinfo functional test

### PR DESCRIPTION
**Pull request by promag**
The `getdescriptorinfo` RPC was added in #15368, this PR adds some tests.

### CODE REVIEW

**Comment by promag** [comment]
@sipa not sure if it makes sense to test these examples. Initially I was only testing the interface.

File: test/functional/test_runner.py [hunk 1]
```diff
@@ -190,6 +190,7 @@
     'rpc_help.py',
     'feature_help.py',
     'feature_shutdown.py',
+    'feature_descriptor.py',
```
Comment (MarcoFalke) [inline test/functional/test_runner.py]: According to the naming guide, this should be rpc_descriptor?

Comment (promag) [reply test/functional/test_runner.py]: Descriptor is not a RPC thing is it?

Comment (Sjors) [reply test/functional/test_runner.py]: Or `rpc_getdescriptorinfo`

**Review by Sjors — APPROVED** [review]
tACK 11778c on macOS 10.14.3.

We should distinguish two reasons for having test descriptors:

1. To test the command output (make sure no fields are missing, all error RPC error conditions are covered)
2. To test `descsum_create` i.e. the Python implementation of descriptor checksums.

(2) probably deserves a seperate test that is called by `test_runner` but otherwise doesn't need any of the framework. On the other hand, just adding it this test file seems fine too.
 
It could be useful to put a bunch of test descriptors in a single JSON file to reuse between tests and to make it easier for [other projects](https://github.com/bitcoin-core/HWI/pull/130) to borrow.

**Comment by jonasschnelli** [comment]
More tests are always good. Thanks for adding.
Concept AcK

**Comment by practicalswift** [comment]
Concept ACK

Thanks for adding!

**Comment by jachiang** [comment]
@promag I looked into this PR in preparation for the PR Review Club IRC session later this week hosted by John and Harding and had a couple questions or thoughts:

1. I believe the RPC test does not cover a case where no checksum is appended to the output descriptor? Currently, `test_desc` will append a checksum to every descriptor literal. Could we extend `test_desc` parameters to allow for such a test:
e.g. `def test_desc(self, desc, *appendchecksum*, isrange, issolvable, hasprivatekeys):`

2. It would seem that we could cover a case with `hasprivatekeys=True`
e.g. `wpkh(WIF)#checksum` or `wsh(multi(1, XPRV/1/0/*, XPUB/0/0/*))#checksum`

Thanks for adding this test.

File: test/functional/test_runner.py (hunk 1 above)
Comment (promag) [reply test/functional/test_runner.py]: Ok, will rename to @Sjors suggestion.

**Comment by promag** [comment]
@jachiang FYI these were taken from `doc/descriptors.md`.

**Review by meshcollider — COMMENTED** [review]
utACK 11778c8fd5cc13fcbcbc746ea1ca7ddef13a7de3

As mentioned above, it would be good to include some tests where `hasprivatekeys=True`

File: test/functional/test_runner.py (hunk 1 above)
Comment (MarcoFalke) [reply test/functional/test_runner.py]: @promag Are you going to rename this or not. If not I will merge as is.

Comment (promag) [reply test/functional/test_runner.py]: Renamed to rpc_getdescriptorinfo.py.

**Pull request merged**